libraries/tentacle.rb

```diff
@@ -41,10 +43,31 @@ def installer_url(version)
       "https://download.octopusdeploy.com/octopus/Octopus.Tentacle.#{version}-x64.msi"
     end
 
+    def tentacle_exists?(server, api_key, thumbprint)
+      ::JSON.parse(api_client(server, api_key).get('/machines/all')).any? do |machine|
```

Comment:
Yeah I agree this would be the best approach with integration from octopus deploys side.  The short term gap would be to save a guard file that can be checked and then skipped without the API requests that are needed.  Since this check is only run once and only if a node is not present this would have a similar effect.  If in the future we wanted to make chef be able to change / keep tags and environments up to date it would make sense to use a more efficient endpoint to find machines.